feat: bounties tab

PeterMPhillips opened the pull request:
This PR adds a Bounties tab to the Projects app:
![Peek 2020-01-20 12-55](https://user-images.githubusercontent.com/19808076/72776706-f8fbbe80-3bc7-11ea-8560-01de757f3523.gif)

It also supports swiping on mobile:
![Peek 2020-01-20 12-56](https://user-images.githubusercontent.com/19808076/72776725-05801700-3bc8-11ea-92ab-c45fec30221f.gif)

Additionally, this PR refactors some of the code that queries the issues data. All the queries have been moved from Issues and Bounties into an IssuesQueryContext. This reduces the frequency that the user will see the 'Loading...' screen when they navigate to the Issues and Bounties tabs.

chadoh commented:
@javieralaves my understanding of this design was that we decided the white background in the columns should only be as tall as the content within that column, which matches how Trello works. I think Peter may have implemented it using an older design, which has the white background extend to the height of the entire page, even when there's no content in the column.

Which approach would you like us to take here?

javieralaves commented:
@chadoh Good point, I didn't update the specifications after our discussion to point that out. If it's a quick-fix to update the issue-responsive column height, let's do it. Otherwise, this is a good iteration that we can improve later on.

PeterMPhillips committed:
feat(projects): bounties tab

In addition to viewing all issues for a project, this allows you to view
all bounties that exist for all projects in an organization

Co-authored-by: Chad Ostrowski <hi@chadoh.com>

chadoh commented:
@PeterMPhillips @javieralaves I've updated this to be compatible with the latest `dev` and made a couple other improvements. Screenshots of dark mode:

<img width="740" alt="Screen Shot 2020-01-23 at 22 10 20" src="https://user-images.githubusercontent.com/221614/73041971-7d9b4680-3e2d-11ea-9eb1-98ab6258a85a.png">

Side scrolling now uses pure CSS instead of a JS plugin:

![bounties side scroll 2020-01-23 at 22 11 45](https://user-images.githubusercontent.com/221614/73041974-81c76400-3e2d-11ea-82f1-fbbef47ea74a.gif)

This works well at first page load, but currently breaks when you resize your window, which is a weird browser bug. I doubt this is important enough to justify the complexity & bundle bloat for a JS plugin, though?

One thing that's currently broken in this new version: navigating back from an issue detail page brings you back to the ProjectDetail for that issue, rather than the Bounties tab. I'd like to implement `goBack` functionality in our `usePathHelpers` hook to address that, which I've done once before and then discarded, so it shouldn't take long. I don't know if it's worth doing as part of this PR or not.

topocount reviewed (changes requested):
@PeterMPhillips @chadoh This is looking really smooth! Bountties requiring applications worked perfectly, but when I created multiple open-submission bounties at once, only one loaded in the Bounties view (at least until I did something that required another query of bounty state, in this case accepting the fulfillment of another bounty). The missing bounty showed up in the issues view no problem. Probably won't be too hard of a fix since bounties with applications are already solid

**EDIT** Neither I, nor others have been able to replicate this aside from the first time mentioned above. If it pops up during UAT we'll look more deeply into it.
